codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -111,6 +197,71 @@ public void updateStreamableBytes(StreamState state) {
     }
 
     @Override
+    public void updateDependencyTree(int childStreamId, int parentStreamId, short weight, boolean exclusive) {
+        if (weight < MIN_WEIGHT || weight > MAX_WEIGHT) {
+            throw new IllegalArgumentException(String.format(
+                    "Invalid weight: %d. Must be between %d and %d (inclusive).", weight, MIN_WEIGHT, MAX_WEIGHT));
+        }
+        if (childStreamId == parentStreamId) {
+            throw new IllegalArgumentException("A stream cannot depend on itself");
+        }
+
+        State state = state(childStreamId);
+        if (state == null) {
+            // If there is no State object that means there is no Http2Stream object and we would have to keep the
+            // State object in the stateOnlyMap and stateOnlyRemovalQueue. However if maxStateOnlySize is 0 this means
+            // stateOnlyMap and stateOnlyRemovalQueue are empty collections and cannot be modified so we drop the State.
+            if (maxStateOnlySize == 0) {
+                return;
+            }
+            state = new State(childStreamId);
+            stateOnlyRemovalQueue.add(state);
+            stateOnlyMap.put(childStreamId, state);
+        }
+
+        State newParent = state(parentStreamId);
+        if (newParent == null) {
+            // If there is no State object that means there is no Http2Stream object and we would have to keep the
+            // State object in the stateOnlyMap and stateOnlyRemovalQueue. However if maxStateOnlySize is 0 this means
+            // stateOnlyMap and stateOnlyRemovalQueue are empty collections and cannot be modified so we drop the State.
+            if (maxStateOnlySize == 0) {
+                return;
+            }
+            newParent = new State(parentStreamId);
+            stateOnlyRemovalQueue.add(newParent);
+            stateOnlyMap.put(parentStreamId, newParent);
+        }
+
+        // if activeCountForTree == 0 then it will not be in its parent's pseudoTimeQueue and thus should not be counted
+        // toward parent.totalQueuedWeights.
+        if (state.activeCountForTree != 0 && state.parent != null) {
+            state.parent.totalQueuedWeights += weight - state.weight;
+        }
+        state.weight = weight;
+
+        if (newParent != state.parent || (exclusive && newParent.children.size() != 1)) {
```

Comment:
It may not be necessary to cache it, and it should be inlined because it is an accessor.